eva-tools/src/main/java/embl/ebi/variation/eva/vcfdump/BiodataVariantToVariantContextConverter.java

```diff
@@ -0,0 +1,183 @@
+
+    private String getContextNucleotideFromCellbaseCachingRegions(Variant variant, int start, Region region) throws CellbaseSequenceDownloadError {
+        if (cellbaseClient != null) {
+            if (regionSequence == null) {
+                // if an indel start is the first nucleotide of the region, we will need the previous nucleotide, so we are adding
+                // the preceding nucleotide to the region (region.getStart()-1)
+                int regionStart = region.getStart() - 1;
+                int regionEnd = region.getEnd();
+                try {
+                    regionSequence = cellbaseClient.getSequence(new Region(variant.getChromosome(), regionStart, regionEnd));
+                } catch (Exception e) {
+                    throw new CellbaseSequenceDownloadError("Error getting from Cellbase sequence for Region " + variant.getChromosome() +
+                            ":" + regionStart + "-" + regionEnd, e);
+                }
+            }
+            String nucleotide = getNucleotideFromRegionSequence(start, region.getStart(), regionSequence);
+            return nucleotide;
+        } else {
+            throw new IllegalArgumentException(String.format(
+                    "CellBase was not provided, needed to fill empty alleles in variant %s:%d:%s>%s",
+                    variant.getChromosome(), variant.getStart(), variant.getReference(), variant.getAlternate()));
+        }
+    }
+
+    private String getNucleotideFromRegionSequence(int start, int regionStart, String regionSequence) {
+        int relativePosition = start - regionStart;
+        return regionSequence.substring(relativePosition, relativePosition + 1);
+    }
+
+    private Set<Genotype> getGenotypes(Variant variant, String[] allelesArray) {
+        Set<Genotype> genotypes = new HashSet<>();
+
+        Allele[] variantAlleles =
+                {Allele.create(allelesArray[0], true), Allele.create(allelesArray[1]), Allele.create(Allele.NO_CALL, false)};
+
+        for (VariantSource source : sources) {
+            List<VariantSourceEntry> variantStudyEntries =
+                    variant.getSourceEntries().values().stream().filter(s -> s.getStudyId().equals(source.getStudyId())).collect(Collectors.toList());
+            for (VariantSourceEntry variantStudyEntry : variantStudyEntries) {
+                genotypes = getStudyGenotypes(genotypes, variantAlleles, variantStudyEntry);
+            }
+        }
+        return genotypes;
+    }
+
+    private Set<Genotype> getStudyGenotypes(Set<Genotype> genotypes, Allele[] variantAlleles, VariantSourceEntry variantStudyEntry) {
+        for (Map.Entry<String, Map<String, String>> sampleEntry : variantStudyEntry.getSamplesData().entrySet()) {
+            String gt = sampleEntry.getValue().get(GENOTYPE_KEY);
+            org.opencb.biodata.models.feature.Genotype genotype = new org.opencb.biodata.models.feature.Genotype(gt, variantAlleles[0].getBaseString(), variantAlleles[1].getBaseString());
```

Comment:
The biodata genotype class was being used in the original dumper code, as a convenience class for parsing the GT string. Now that we are not using phased genotypes, this could be replaced by a regular expression, to make the code more readable.